Description: Avoid build problem on hurd
 Hurd's bits/errno.h defines constant ED, so rename
 a local variable to avoid the clash.
Author: Pino Toscano <pino@kde.org>
Bug: 552383
Forwarded: not yet


--- boost1.40-1.40.0.orig/boost/math/special_functions/ellint_rd.hpp
+++ boost1.40-1.40.0/boost/math/special_functions/ellint_rd.hpp
@@ -29,7 +29,7 @@
 T ellint_rd_imp(T x, T y, T z, const Policy& pol)
 {
     T value, u, lambda, sigma, factor, tolerance;
-    T X, Y, Z, EA, EB, EC, ED, EE, S1, S2;
+    T X, Y, Z, EA, EB, EC, ED_, EE, S1, S2;
     unsigned long k;
 
     BOOST_MATH_STD_USING
@@ -93,9 +93,9 @@
     EA = X * Y;
     EB = Z * Z;
     EC = EA - EB;
-    ED = EA - 6 * EB;
-    EE = ED + EC + EC;
-    S1 = ED * (ED * T(9) / 88 - Z * EE * T(9) / 52 - T(3) / 14);
+    ED_ = EA - 6 * EB;
+    EE = ED_ + EC + EC;
+    S1 = ED_ * (ED_ * T(9) / 88 - Z * EE * T(9) / 52 - T(3) / 14);
     S2 = Z * (EE / 6 + Z * (-EC * T(9) / 22 + Z * EA * T(3) / 26));
     value = 3 * sigma + factor * (1 + S1 + S2) / (u * sqrt(u));
 
